src/components/VideoPreview.tsx

```diff
@@ -1,33 +1,49 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { useVideo } from '@/api/hooks'
 import routes from '@/config/routes'
-import VideoPreviewBase, { VideoPreviewBaseMetaProps } from '@/shared/components/VideoPreviewBase/VideoPreviewBase'
+import VideoPreviewBase, {
+  VideoPreviewBaseMetaProps,
+  VideoPreviewPublisherProps,
+} from '@/shared/components/VideoPreviewBase/VideoPreviewBase'
+import { useDrafts } from '@/hooks'
 
-type VideoPreviewProps = {
+export type VideoPreviewProps = {
   id?: string
   progress?: number
+  isLoading?: boolean
   className?: string
-} & VideoPreviewBaseMetaProps
+} & VideoPreviewBaseMetaProps &
+  VideoPreviewPublisherProps
 
-const VideoPreview: React.FC<VideoPreviewProps> = ({ id, progress, className, ...metaProps }) => {
+const VideoPreview: React.FC<VideoPreviewProps> = ({ id, className, isLoading = false, ...metaProps }) => {
+  const [loadOnMount, setloadOnMount] = useState(true)
+  const { drafts } = useDrafts('video')
   const { video, loading } = useVideo(id ?? '', { fetchPolicy: 'cache-first', skip: !id })
-  const isLoading = loading || id === undefined
+
+  useEffect(() => {
+    setloadOnMount(false)
+  }, [])
+
+  const _isLoading = loading || id === undefined || isLoading || loadOnMount
```

Comment:
Unless necessary in the context, let's try to avoid direct comparison to `undefined`. We can do `!id` instead